packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -217,11 +217,22 @@ class EditableTextState extends State<EditableText> implements TextInputClient {
   TextSelectionOverlay _selectionOverlay;
 
   final ScrollController _scrollController = new ScrollController();
+  bool _didAutoFocus = false;
 
   @override
   void initState() {
     super.initState();
     _currentValue = config.value;
+    config.focusNode.addListener(_handleFocusChanged);
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (!_didAutoFocus && config.autofocus) {
+      FocusScope.of(context).autofocus(config.focusNode);
+      _didAutoFocus = true;
+    }
```

Comment:
We discussed this in person. For the record, my concern was that the widget tree and the focus tree should remain similar, since we almost always end up with bugs when we have two trees that are related to each other but they spend time nonisomorphic.